open-craft/opencraft
# [BB-3616]: Add check for DNS configuration

If external domain is added to the instance, we need to check that proper CNAME records are created with the DNS by use.

**Related Tickets**: [BB-3616](https://tasks.opencraft.com/browse/BB-3616)

**Testing Instructions**:
1. Checkout to this branch
2. Run migrations using `make migrate`
2. Run dev server using `make run.dev`
3. Open admin panel
4. Create a test `User` with a `BetaTestApplication`
5. Make the following curl request (replace variable with proper values)

```
curl -X PATCH 'http://localhost:5000/api/v2/instances/openedx_config/:instance_id/' \
  -H 'Authorization: Bearer <jwt-token>' \
  -H 'Content-Type: application/json' \
  -H 'Accept: */*' \
  --data-raw '{"external_domain":"a-new-domain.com"}' \
  --compressed
```
6. Verify that a task is scheduled (worker logs)
7. Verify that the task is started after 1 minute (worker logs)
8. Verfiy that the DNS configuration check fails and is logged (worker logs)
9. Verfiy that the Email log is printed (worker logs)

2. Check that `templates/emails/dns_not_configured.txt` and `templates/emails/dns_not_configured.html` have appropriate data corresponding to [this Zeplin design](https://app.zeplin.io/project/5ea850e04326be273e4bcf5a/screen/5fdb0719a491febb0d11d5a2)
3. Run `make test.one registration.tests` and verfiy that all unittests pass.
